MovieApp/MovieApp/App/Features/MovieBrowser/Components/DetailScreen/TitleView/DetailTitleView+Design.swift

```diff
@@ -0,0 +1,118 @@
+import UIKit
+
+extension DetailTitleView: DesignProtocol {
+    
+    func buildViews() {
+        createViews()
+        styleViews()
+        defineLayoutForViews()
+    }
+    
+    func createViews() {
+        backgroundImageView = UIImageView()
+        addSubview(backgroundImageView)
+        
+        gradientView = UIView()
+        addSubview(gradientView)
+        
+        titleLabel = UILabel()
+        addSubview(titleLabel)
+        
+        yearLabel = UILabel()
+        addSubview(yearLabel)
+        
+        releaseDateLabel = UILabel()
+        addSubview(releaseDateLabel)
+        
+        genresLabel = UILabel()
+        addSubview(genresLabel)
+        
+        durationLabel = UILabel()
+        addSubview(durationLabel)
+        
+        favouritesButton = FavouriteButton()
+        addSubview(favouritesButton)
+        
+        progressBar = ProgressBarView(percentage: details.userScore)
+        addSubview(progressBar)
+    }
+    
+    func styleViews() {
+        backgroundImageView.kf.setImage(with: details.posterPath)
+        backgroundImageView.contentMode = .scaleAspectFill
+        backgroundImageView.clipsToBounds = true
+        
+        let gradientLayer = CAGradientLayer()
+        gradientLayer.frame = CGRect(x: 0, y: 0, width: Int(UIScreen.main.bounds.width) + 1, height: DetailTitleView.height)
+        gradientLayer.colors = [UIColor.clear.cgColor, UIColor.black.cgColor]
+        gradientView.layer.insertSublayer(gradientLayer, at: 0)
+                
+        titleLabel.text = details.title
+        titleLabel.textColor = .white
+        titleLabel.font = .heading2
+        
+        yearLabel.text = "(\(details.year))"
+        yearLabel.textColor = .white
+        yearLabel.font = .heading2regular
+        
+        releaseDateLabel.text = details.releaseDate
+        releaseDateLabel.textColor = .white
+        releaseDateLabel.font = .regularMedium
+        
+        genresLabel.text = details.genres.joined(separator: ", ")
+        genresLabel.textColor = .white
+        genresLabel.font = .regularMedium
+        
+        durationLabel.text = details.duration
+        durationLabel.textColor = .white
+        durationLabel.font = .smallBold
+        
+        progressBar.progressAnimation(duration: 1.2)
+    }
+    
+    func defineLayoutForViews() {
+        backgroundImageView.snp.makeConstraints {
+            $0.edges.equalToSuperview()
+        }
+        
+        gradientView.snp.makeConstraints {
+            $0.edges.equalTo(backgroundImageView.snp.edges)
+        }
+                
+        favouritesButton.snp.makeConstraints {
+            $0.bottom.equalToSuperview().inset(5 * offset)
+            $0.leading.equalToSuperview().offset(4 * offset)
+        }
+        
+        genresLabel.snp.makeConstraints {
+            $0.bottom.equalTo(favouritesButton.snp.top).offset(-4 * offset)
```

Comment:
When creating constraints you should write them in order top->bottom and left -> right.
That's because UIKit lays out the views in that order and you have to "hack" offsets with negative values. All constraints should be `top.equalTo(.bottom).offset(x)` and leading to trailing.
I know you're anchoring everything to the bottom and it was easier to write this way, but please try, regardless of the anchor point, write the constraints in that order.

Reply:
[Here](https://github.com/lihter/Movie-App/pull/28/commits/3a1dd8418b7535a8fac5faf7eea2183f097a6eef#diff-8dc93338bfb7c8f14d32149c2a00bb34461a90ec91068f6510b69818f73e4f56R82-R123)